portfolio/src/main/webapp/js/main.js

```diff
@@ -244,3 +256,39 @@ function indexOfMax(arr) {
 
     return maxIndex;
 }
+
+/**
+  * Responds to the mouse-in event on a map shape (state).
+  *
+  * @param {?google.maps.MouseEvent} e
+  */
+function mouseInToRegion(e) {
+  // set the hover state so the setStyle function can change the border
+  e.feature.setProperty('state', 'hover');
+
+  // update the label
+  document.getElementById('data-label').textContent = e.feature.getProperty('NAME');
```

Comment:
Consider refactor this block of code to a for loop for better readability since they are following the same pattern.